modules/rules_go/0.33.0/MODULE.bazel

```diff
@@ -0,0 +1,23 @@
+module(
+    name = "rules_go",
+    version = "0.33.0",
+    compatibility_level = 0,
+    toolchains_to_register = [
+        "@go_default_sdk//:all",
+    ],
+)
+
+print("WARNING: The rules_go Bazel module is still highly experimental and subject to change at any time. Only use it to try out bzlmod for now.")
+
+bazel_dep(name = "bazel_skylib", version = "1.2.0")
+bazel_dep(name = "platforms", version = "0.0.4")
+
+non_module_dependencies = use_extension("@rules_go//go/private:extensions.bzl", "non_module_dependencies")
```

Comment:
just a comment: you should be able to just write `use_extension("//go/private...")` here, which also removes some of the fuss in the patch file. (This might not be applicable everywhere in this PR yet, as you are probably already aware :)